### PR TITLE
Fixed $export-poll-status null point exception issue

### PR DESCRIPTION
**Pull request by Qingyixia**
#4427

### CODE REVIEW

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderTest.java [hunk 1]
```diff
@@ -923,6 +923,74 @@ public void testGetBulkExport_outputFormat_FhirNdJson_inUrl() throws IOException
...
+		input.addParameter(JpaConstants.PARAM_EXPORT_POLL_STATUS_JOB_ID, new StringType(A_JOB_ID));
+
+		// Initiate Export Poll Status
+		HttpPost post = new HttpPost("http://localhost:" + myPort + "/" + JpaConstants.OPERATION_EXPORT_POLL_STATUS);
+		post.addHeader(Constants.HEADER_PREFER, Constants.HEADER_PREFER_RESPOND_ASYNC);
+		post.setEntity(new ResourceEntity(myCtx, input));
+
+		try (CloseableHttpResponse response = myClient.execute(post)) {
+			ourLog.info("Response: {}", response.toString());
+			assertEquals(202, response.getStatusLine().getStatusCode());
```
Comment (lukedegruchy) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderTest.java:971]: Nitpick:  try to use constant for 202 and Accepted if you can find them in libraries.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java [hunk 2]
```diff
@@ -291,7 +294,24 @@ public void exportPollStatus(
...
+					.filter(param -> param.getName().equals(JpaConstants.PARAM_EXPORT_POLL_STATUS_JOB_ID))
+					.iterator().next().getValue();
+			} catch (NoSuchElementException e) {
+				throw new InvalidRequestException(Msg.code(2227) + "$export-poll-status requires a job ID, please provide the value of jobId!");
+			}
+		}
+
 		Batch2JobInfo info = myJobRunner.getJobInfo(theJobId.getValueAsString());
+		if (info == null) {
+			throw new ResourceNotFoundException(Msg.code(2040) + "Unknown instance ID: " + theJobId);
```
Comment (lukedegruchy) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:313]: Please add a more descriptive error message here.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java [hunk 3]
```diff
@@ -291,7 +294,24 @@ public void exportPollStatus(
...
 		theRequestDetails.getServer().addHeadersToResponse(response);
+
+		// When export-poll-status through POST
+		// Get theJobId from the request details
+		if (theJobId == null){
+			try {
+				Parameters parameters = (Parameters) theRequestDetails.getResource();
+				theJobId = (IPrimitiveType<String>) parameters.getParameter().stream()
+					.filter(param -> param.getName().equals(JpaConstants.PARAM_EXPORT_POLL_STATUS_JOB_ID))
+					.iterator().next().getValue();
```
Comment (JasonRoberts-smile) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:305]: This is fine, but for your edification, it is also possible to replace this line with `.findFirst().orElseThrow(() -> new InvalidRequestException(theMessage))`, which does the same thing but is a bit more readable.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java [hunk 4]
```diff
@@ -291,7 +294,24 @@ public void exportPollStatus(
...
+			} catch (NoSuchElementException e) {
+				throw new InvalidRequestException(Msg.code(2227) + "$export-poll-status requires a job ID, please provide the value of jobId!");
```
Comment (JasonRoberts-smile) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:307]: nitpick: It's generally best not to end an error message with an exclamation. The client might think you are angry at them for making a mistake in their call.

**Review by JasonRoberts-smile — APPROVED** [review]
approved with comments

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java (hunk 3 above)
Comment (Qingyixia) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/bulk/export/provider/BulkDataExportProvider.java:305]: That's much better! Thanks for the suggestion.

**Pull request merged**